mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -532,28 +539,14 @@ class Word2VecModel private[spark] (
     blas.sgemv(
       "T", vectorSize, numWords, alpha, wordVectors, vectorSize, fVector, 1, beta, cosineVec, 1)
 
-    // Need not divide with the norm of the given vector since it is constant.
     val cosVec = cosineVec.map(_.toDouble)
-    var ind = 0
-    val vecNorm = blas.snrm2(vectorSize, fVector, 1)
```

Comment:
Removing this means that passing in a vector that is not in the vocab will not give the expected result as it is not explicitly normalized. As an example, some applications may pass in the average vector for a set of words to compute synonyms.